cli/tests/pcluster/utils/test_pcluster_utils.py

```diff
@@ -0,0 +1,229 @@
+    cfn_params = [{"ParameterKey": "Key", "ParameterValue": "Value"}]
+    expected_params = {
+        "StackName": stack_name,
+        "TemplateBody": json.dumps(template_body, indent=2),
+        "Parameters": cfn_params,
+    }
+    emsg = "An error occurred (ValidationError) when calling the UpdateStack operation: No updates are to be performed."
+    mocked_requests = [MockedBoto3Request(method="update_stack", response=emsg, expected_params=expected_params)]
+    boto3_stubber("cloudformation", mocked_requests, generate_errors=True)
+    assert_that(utils.update_stack_template(stack_name, template_body, cfn_params)).is_none()
+
+
+def test_update_stack_error_not_ignored(boto3_stubber):
+    """Verify no exception is raised when update_stack is called with a template equal to the stack's current one."""
+    stack_name = "stack_name"
+    template_body = {"TemplateKey": "TemplateValue"}
+    cfn_params = [{"ParameterKey": "Key", "ParameterValue": "Value"}]
+    expected_params = {
+        "StackName": stack_name,
+        "TemplateBody": json.dumps(template_body, indent=2),
+        "Parameters": cfn_params,
+    }
+    emsg = "Some strange error occurred."
+    mocked_requests = [MockedBoto3Request(method="update_stack", response=emsg, expected_params=expected_params)]
+    boto3_stubber("cloudformation", mocked_requests, generate_errors=True)
+    expected_message = "Unable to update stack template for stack {stack_name}: {emsg}".format(
+        stack_name=stack_name, emsg=emsg
+    )
+    with pytest.raises(SystemExit, match=expected_message):
+        utils.update_stack_template(stack_name, template_body, cfn_params)
+
+
+def test_get_cluster_substacks(mocker):
+    """Verify that utils.get_cluster_substacks returns the correct stacks."""
+    stack_type = "AWS::CloudFormation::Stack"
+    resources = [
+        {"ResourceType": "Not a stack", "ResourceName": "name_one"},
+        {"ResourceType": stack_type, "ResourceName": "name_two"},
+        {"ResourceType": "Also not a stack", "ResourceName": "name_three"},
+        {"ResourceType": stack_type, "ResourceName": "name_four"},
+    ]
+    mocker.patch("pcluster.utils.get_stack_resources").return_value = resources
+    cluster_name = "cluster_name"
+    stack_name = utils.get_stack_name(cluster_name)
+    expected_substacks = [d for d in resources if d.get("ResourceType") == stack_type]
+    observed_substacks = utils.get_cluster_substacks(cluster_name)
+    utils.get_stack_resources.assert_called_with(stack_name)
+    assert_that(observed_substacks).is_equal_to(expected_substacks)
+
+
+def test_get_cloudwatch_logs_substack(mocker):
+    """Verify that utils.get_cloudwatch_logs_substack returns the expected stack."""
+    cluster_name = "cluster_name"
+    cw_logs_substack = {
+        "StackName": "{stack_name}-{substack_prefix}-".format(
+            stack_name=utils.get_stack_name(cluster_name), substack_prefix=utils.CW_LOGS_SUBSTACK_PREFIX
```

Comment:
I'd use the real string in place of CW_LOGS_SUBSTACK_PREFIX.
In this way this will became a double check: for the get_cloudwatch_logs_substack function and for the prefix itself.

Reply:
Done.